libs/dh/market-participant/data-access-api/src/lib/dh-market-participant-edit-actor-data-access-api.store.ts

```diff
@@ -148,16 +134,88 @@ export class DhMarketParticipantEditActorDataAccessApiStore extends ComponentSto
     )
   );
 
+  readonly loadInitialData = this.effect(
+    (routeParams$: Observable<{ organizationId: string; actorId: string }>) =>
+      routeParams$.pipe(
+        switchMap((x) =>
+          forkJoin({
+            gridAreas: this.getGridAreas(),
+            actorAndContacts: this.getActorAndContacts(x),
+          }).pipe(
+            tap((x) =>
+              this.patchState({
+                gridAreaChanges: x.gridAreas.filter((y) =>
+                  x.actorAndContacts.gridAreas.includes(y.id)
+                ),
+              })
+            )
+          )
+        )
+      )
+  );
+
+  readonly getGridAreas = () =>
+    of('').pipe(
+      tap(() => this.patchState({ isLoadingGridAreas: true })),
+      switchMap(() =>
+        this.gridAreaHttpClient
+          .v1MarketParticipantGridAreaGet()
+          .pipe(
+            tap((gridAreas) =>
+              this.patchState({
+                gridAreas: gridAreas.sort((a, b) =>
+                  a.code.localeCompare(b.code)
+                ),
+              })
+            )
+          )
+          .pipe(catchError(this.handleError))
+      ),
+      tap(() => this.patchState({ isLoadingGridAreas: false }))
+    );
+
+  readonly getActorAndContacts = (routeParams: {
+    organizationId: string;
+    actorId: string;
+  }) =>
+    // contacts not implemented yet.
+    of(routeParams).pipe(
+      tap(() => this.patchState({ isLoadingActor: true })),
+      switchMap((routeParams) => {
+        if (!routeParams.actorId) {
+          this.patchState({
+            isLoadingActor: false,
+            organizationId: routeParams.organizationId,
+          });
+          return EMPTY;
```

Comment:
question: does `forkJoin` in `loadInitialData` still work in case `EMPTY` is returned? Don't think the `tap` operator on line 145 is called in that case.

Reply:
fixed @dzhavat